linera-service/src/linera.rs

```diff
@@ -1655,14 +1654,13 @@ impl Runnable for Job {
             }
 
             Watch { chain_id, raw } => {
-                let chain_client = context.make_chain_client(storage, chain_id);
+                let mut chain_client = context.make_chain_client(storage, chain_id);
                 let chain_id = chain_client.chain_id();
                 info!("Watching for notifications for chain {:?}", chain_id);
-                let mut tracker = NotificationTracker::default();
-                let mut notification_stream =
-                    ChainClient::listen(Arc::new(Mutex::new(chain_client))).await?;
+                let mut notification_stream = chain_client.subscribe().await?;
+                ChainClient::listen(Arc::new(Mutex::new(chain_client))).await?;
                 while let Some(notification) = notification_stream.next().await {
-                    if raw || tracker.insert(notification.clone()) {
+                    if raw {
                         println!("{:?}", notification);
```

Comment:
By the way, this should be either a log `info!("{:?}", notification)` or printing JSON `println!("{}", serde_json::to_string(notification))`

Reply:
I was going to suggest we remove this command entirely but then realized this may be the command for chain leaders on validators.

Reply:
Not sure: Chain leaders on validators would not only watch, but also create blocks, so it would probably be another command?
I'll make it JSON for now; if we remove it I'd do that in a separate PR.